### DISCUSSION TITLE
NVME SSD Settings

### DISCUSSION THREAD

**Discussion by JohnDotOwl** [le0pard/pgtune#48] [Ideas]
**Is your feature request related to a problem? Please describe.**
I'm running postgresql off a NVME Flash drive and it's starting to be really common.

**Describe the solution you'd like**
Add : Data Storage Type - "NVME SSD" under Data Store

**Additional context**
Which configuration should i look into if i would like to make use of my high IOPS nvme drive. 
- High Read up to 3500MB/s 
- High IOPS - 600,000 IOPS 

Which settings should I look into if i would like to reduce the number of writes done to the disk as flash storage life span actually depends on amount of writes done to the disk.

**Comment by le0pard** [comment]
> Which configuration should i look into if i would like to make use of my high IOPS nvme drive

> Which settings should I look into if i would like to reduce the number of writes done to the disk as flash storage life span actually depends on amount of writes done to the disk.

This definitions exists on different sides - do you want provide PostgreSQL ability to use NVME drive or do you want safe it lifespan and not allow DB write to it too much?

Fast disks will only help if your workload is I/O bound, and in your case it will at best improve speed by a factor of six. Creating the proper index can easily improve performance by a factor of 1000.

To make the best of your investment, set `effective_io_concurrency` and `random_page_cost` to reflect the physical properties of your storage, so that PostgreSQL can make the best of it. What values to set? You should do tests directly on your system with NVME to find best values, which will work for you PostgreSQL workload (not only based on hardware). 

Pgtune can give you good initial settings, which you can select as "Network (SAN) storage" and continue tuning yourself for your system.
